synapse/module_api/__init__.py

```diff
@@ -1226,6 +1229,43 @@ async def sleep(self, seconds: float) -> None:
 
         await self._clock.sleep(seconds)
 
+    async def send_http_push_notification(
+        self,
+        user_id: str,
+        device_id: Optional[str],
+        content: JsonDict,
+        tweaks: JsonMapping = {},
+    ) -> bool:
+        """Send an HTTP push notification that is forwarded to the registered push gateway
+        for the specified user/device.
+
+        Added in Synapse v1.82.0.
+
+        Args:
+            user_id: The user ID to send the push notification to.
+            device_id: The device ID of the device where to send the push notification. If `None`,
+            the notification will be sent to all registered HTTP pushers of the user.
+            content: A dict of values that will be put in the `notification` field of the push
+            (cf Push Gatway spec). `devices` field will be overrided if included.
+            tweaks: A dict of `tweaks` that will be inserted in the `devices` section, cf spec.
+
+        Returns:
+            True if at least one push was succesfully sent, False if no matching pusher has been
+            found or an error occured when sending the push.
```

Comment:
Perhaps we should differentiate between no pusher found and errors. Also currently if we don't specify `device_id` and some pusher fails but at least one succeeded, this is still returning success.

Reply:
Maybe return a map of `device_id -> success`?